scipy/optimize/shgo_m/sobol_seq.py

```diff
@@ -0,0 +1,371 @@
+            self.v = np.zeros((self.dim_max, self.log_max))
+            self.v[0:40, 0] = np.transpose([
+                1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
+
+            self.v[2:40, 1] = np.transpose([
+                1, 3, 1, 3, 1, 3, 3, 1,
+                3, 1, 3, 1, 3, 1, 1, 3, 1, 3,
+                1, 3, 1, 3, 3, 1, 3, 1, 3, 1,
+                3, 1, 1, 3, 1, 3, 1, 3, 1, 3])
+
+            self.v[3:40, 2] = np.transpose([
+                7, 5, 1, 3, 3, 7, 5,
+                5, 7, 7, 1, 3, 3, 7, 5, 1, 1,
+                5, 3, 3, 1, 7, 5, 1, 3, 3, 7,
+                5, 1, 1, 5, 7, 7, 5, 1, 3, 3])
+
+            self.v[5:40, 3] = np.transpose([
+                1, 7, 9, 13, 11,
+                1, 3, 7, 9, 5, 13, 13, 11, 3, 15,
+                5, 3, 15, 7, 9, 13, 9, 1, 11, 7,
+                5, 15, 1, 15, 11, 5, 3, 1, 7, 9])
+
+            self.v[7:40, 4] = np.transpose([
+                9, 3, 27,
+                15, 29, 21, 23, 19, 11, 25, 7, 13, 17,
+                1, 25, 29, 3, 31, 11, 5, 23, 27, 19,
+                21, 5, 1, 17, 13, 7, 15, 9, 31, 9])
+
+            self.v[13:40, 5] = np.transpose([
+                37, 33, 7, 5, 11, 39, 63,
+                27, 17, 15, 23, 29, 3, 21, 13, 31, 25,
+                9, 49, 33, 19, 29, 11, 19, 27, 15, 25])
+
+            self.v[19:40, 6] = np.transpose([
+                13,
+                33, 115, 41, 79, 17, 29, 119, 75, 73, 105,
+                7, 59, 65, 21, 3, 113, 61, 89, 45, 107])
+
+            self.v[37:40, 7] = np.transpose([
+                7, 23, 39])
+
+            #  Set POLY.
+            self.poly = [
+                1, 3, 7, 11, 13, 19, 25, 37, 59, 47,
+                61, 55, 41, 67, 97, 91, 109, 103, 115, 131,
+                193, 137, 145, 143, 241, 157, 185, 167, 229, 171,
+                213, 191, 253, 203, 211, 239, 247, 285, 369, 299]
+
+            self.atmost = 2 ** self.log_max - 1
+
+            #  Find the number of bits in ATMOST.
+            self.maxcol = self.i4_bit_hi1(self.atmost)
+
+            #  Initialize row 1 of V.
+            self.v[0, 0:self.maxcol] = 1
+
+        # Things to do only if the dimension changed.
+        if dim_num != self.dim_num_save:
+            self.dim_num_save = dim_num
+
+            #  Initialize the remaining rows of V.
+            for i in range(2, dim_num + 1):
+
+                # The bits of the integer POLY(I) gives the form of
+                # self.polynomial I.
+                # Find the degree of self.polynomial I from binary encoding.
+                j = self.poly[i - 1]
+                m = 0
+                j //= 2
+                while j > 0:
+                    j //= 2
+                    m += 1
+
+                # Expand this bit pattern to separate
+                # components of the logical array INCLUD.
+                j = self.poly[i - 1]
```

Comment:
If I understand correctly this part is taking the last m bits and converting to an array, 

```python
includ = [int(x) for x in np.binary_repr(poly[i-1])[-m:]]
```
would do the same. I still don't know why we are looping over poly though.